includes/compat/class-wc-stripe-subs-compat.php

```diff
@@ -336,6 +322,8 @@ public function create_and_confirm_intent_for_renewal( $amount, $order, $prepare
 
 		$intent = WC_Stripe_API::request( $request, 'payment_intents' );
 		if ( ! empty( $intent->error ) ) {
+			// Save the intent ID to the order, even if it failed
+			$this->save_intent_to_order( $order, $intent->error->payment_intent );
```

Comment:
@v18 You pointed out that we should store the intent ID even after an error, I needed to make this change for the flow to work anyway.